src/Chapter2.hs

```diff
@@ -728,7 +736,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat [replicate x x | x <- l]
```

Comment:
Very neat implementation!

You also could just use `concatMap` instead of list comprehensions. Compiler would be able to optimize it better 🙂 

```suggestion
smartReplicate = concatMap (\x -> replicate x x)
```